### PR TITLE
feat: THES-105: Payment State Polling

### PR DESCRIPTION
**Pull request by grmartin**
- feat: THES-105: Payment State Polling
    Payment State polling on Submit Payment and Get Active Order, with a normally non-closeable Dialog.
- fix: Advisories permitted for the moment
    This spawned REV-3598.
- fix: Removal of PAYMENT_STATE.PROCESSING, this state is internal to the Titan backend.

### CODE REVIEW

File: src/payment/data/sagas.js [hunk 1]
```diff
@@ -290,6 +303,53 @@ export function* handleSubmitPayment({ payload }) {
...
+      if (!(yield select(state => state.payment.basket.paymentStatePolling.keepPolling))) {
+        keepPolling = false;
+        yield put(updatePaymentState.fulfill());
+      } else {
+        yield delay(SECS_AS_MS(delaySecs));
+      }
+    } catch (error) {
+      // We dont quit on error.
+      // yield call(handleErrors, error, true);
+      yield put(paymentStateDataReceived({ state: PAYMENT_STATE.HTTP_ERROR }));
```
Comment (grmartin) [inline src/payment/data/sagas.js:347]: Add common logging for errors from New Relic.

File: src/payment/data/actions.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const fetchActiveOrder = createRoutine('FETCH_ACTIVE_ORDER');
 export const addCoupon = createRoutine('ADD_COUPON');
 export const removeCoupon = createRoutine('REMOVE_COUPON');
 export const updateQuantity = createRoutine('UPDATE_QUANTITY');
+export const updatePaymentState = createRoutine('UPDATE_PAYMENT_STATE');
```
Comment (grmartin) [inline src/payment/data/actions.js:23]: name it poll payment state.

File: src/payment/data/sagas.js [hunk 3]
```diff
@@ -290,6 +303,53 @@ export function* handleSubmitPayment({ payload }) {
...
+    try {
+      const basketId = yield select(state => state.payment.basket.basketId);
+      const paymentNumber = yield select(state => (state.payment.basket.payments.length === 0
+        ? null : state.payment.basket.payments[0].paymentNumber));
+
+      if (!basketId || !paymentNumber) {
+        // This shouldn't happen.
+        //   I don't think we need to banner... shouldn't our parent calls recover? (They invoke this)
+        keepPolling = false;
+        yield put(updatePaymentState.fulfill());
```
Comment (grmartin) [inline src/payment/data/sagas.js:331]: throw exception?

File: src/payment/data/sagas.js (hunk 1 above)
Comment (grmartin) [reply src/payment/data/sagas.js:347]: potentially HTTP errors should fll through.

**Review by pshiu — APPROVED** [review]

**Comment by grmartin** [comment]
Splitting exception handling in to another ticket.

**Pull request merged**